packages/realm-web/src/App.ts

```diff
@@ -0,0 +1,239 @@
+            throw new Error("Expected an refresh token in the response");
+        }
+        // Create the user
+        const handle = User.create({
+            app: this,
+            id: userId,
+            accessToken,
+            refreshToken
+        });
+        // If neeeded, fetch and set the profile on the user
+        if (fetchProfile) {
+            const profile = await this.baseTransport.fetch(
+                { method: "GET", path: "/auth/profile" },
+                handle.user
+            );
+            handle.controller.setProfile(profile);
+        }
+        // Add the user at the top of the stack
+        this.users.splice(0, 0, handle);
+        // Return the user
+        return handle.user;
+    }
+
+    /**
+     * Log out a user
+     * @param userOrId The user or id of the user to log out (default: currentUser)
+     */
+    public async logOut(
+        userOrId: Realm.User | string | null = this.currentUser
+    ) {
+        const { user, controller } = this.getUserHandle(userOrId);
+        // Invalidate the refresh token
+        await this.baseTransport.fetch({
+            method: "DELETE",
+            path: "/auth/session",
+            headers: {
+                Authorization: `Bearer ${user.refreshToken}`
+            }
+        });
+        // Make the user forget its tokens
+        controller.forgetTokens();
+        // Set the state of the user
+        controller.setState(UserState.LoggedOut);
+    }
+
+    /**
+     * Remove a user entirely from the app (logs out the user if they're not already logged out)
+     * @param userOrId The user or id of the user to remove.
+     */
+    public async removeUser(userOrId: Realm.User | string) {
+        const { user, controller } = this.getUserHandle(userOrId);
+        // If active - log out the user
+        if (user.state === UserState.Active) {
+            await this.logOut(user);
+        }
+        // Set the state of the user
+        controller.setState(UserState.Removed);
+        // Remove the user from the list of users
+        const index = this.users.findIndex(({ user: u }) => u === user);
+        this.users.splice(index, 1);
+        // TODO: Delete any data / tokens which were persisted
+    }
+
+    /**
+     * The currently active user (or null if no active users exists)
+     */
+    public get currentUser(): Realm.User | null {
+        const activeUserHandles = this.users.filter(
+            ({ user }) => user.state === UserState.Active
+        );
+        if (activeUserHandles.length === 0) {
+            return null;
+        } else {
+            // Current user is the top of the stack
+            return activeUserHandles[0].user;
+        }
+    }
+
+    /**
+     * All active and logged-out users:
+     *  - First in the list are active users (ordered by most recent call to switchUser or login)
+     *  - Followed by logged out users (also ordered by most recent call to switchUser or login).
+     */
```

Comment:
It's the rule object store uses (ask @jsflax) :)

Reply:
@cmelchior and @jsflax Is this sorting behavior the same for the other sdks. Should we align on it?

Reply:
I think Java is not doing that. Based on this: https://github.com/realm/realm-java/blob/5e64f8f2147875bd1496d09c3839a63e9e09420c/realm/realm-library/src/objectServer/java/io/realm/RealmApp.java#L134

Reply:
I clarified the sorting with @jsflax before implementing this, I don't see a reason to change it here. Perhaps theres a discrepancy but let's resolve that in a followup PR in the project where it turns out we need to do it differently.

Reply:
It's not a big deal for me. I just noticed the bahavior. Its still seems doing too much like we are sorting by user state  but also why not sort by first name here. I think sorting should be only part when UI is involved in this case. Lets ignore this.

Reply:
Java just returns the values as provided by Object Store. I don't know how that is implemented. But in either case, I don't have strong opinions either way. 

Ideally, it should probably have been the same, but 🤷

Reply:
I am just afraid of providing such implicit hidden behaviors when we can do the simplest way of returning what we have in the whatever order we have it. It's no big deal for me either but it may pile up after a while and then people start relying on this and may lead to bugs if there are more states in the future. Maybe not in this case specifically but in general.

Reply:
I agree with your sentiment to an extent, and there will be conversations going forward to address this.

The idea is to always keep the "freshest" users on top, organised by user state. As long as it is documented, I do not consider this hidden behaviour.

However, I agree that it is odd. This is something we should reinvestigate in the coming months, but for now, the project was to reimplement the same behaviours the legacy Stitch SDKs had.

Reply:
oh I see. if the legacy Stitch SDK has this then we should leave it as it is.